core/src/main/java/org/infinispan/topology/ClusterTopologyManagerImpl.java

```diff
@@ -449,10 +449,17 @@ private void confirmMembersAvailable() throws Exception {
     *
     * @throws TimeoutException if the timeout expired.
     */
-   private void waitForView(int joinerViewId, long timeout) throws InterruptedException {
-      if (joinerViewId > viewId) {
-         log.tracef("Received a cache topology command with a higher view id: %s, our view id is %s",
-               joinerViewId, viewId);
+   private void waitForJoinerView(Address joiner, int joinerViewId, long timeout)
+         throws InterruptedException {
+      if (joinerViewId > viewId || clusterManagerStatus == ClusterManagerStatus.RECOVERING_CLUSTER) {
+         if (trace) {
+            if (joinerViewId > viewId) {
+               log.tracef("Waiting to install view %s before processing join request from %s", joiner,
```

Comment:
tracef parameters in the wrong order (first if the view, then the joiner address)